rpcs3/Emu/RSX/GL/GLGSRender.cpp

```diff
@@ -832,38 +838,47 @@ bool GLGSRender::load_program()
 		auto mapping = m_uniform_ring_buffer.alloc_from_reserve(fragment_constants_size, m_uniform_buffer_offset_align);
 		fragment_constants_offset = mapping.second;
 
-		static const __m128i mask = _mm_set_epi8(
-			0xE, 0xF, 0xC, 0xD,
-			0xA, 0xB, 0x8, 0x9,
-			0x6, 0x7, 0x4, 0x5,
-			0x2, 0x3, 0x0, 0x1);
-
-		auto ucode = (const rsx::fragment_program::ucode_instr *)info.fragment_shader.decompiled->raw->ucode_ptr;
+		if (!info.consts_initialized || fragment_constants_size != info.fragment_shader.const_buf.size())
+		{
+			info.fragment_shader.const_buf.resize(fragment_constants_size);
 
-		auto dst = (const rsx::fragment_program::ucode_instr *)mapping.first;
+			static const __m128i mask = _mm_set_epi8(
+				0xE, 0xF, 0xC, 0xD,
+				0xA, 0xB, 0x8, 0x9,
+				0x6, 0x7, 0x4, 0x5,
+				0x2, 0x3, 0x0, 0x1);
 
-		for (const auto& constant : info.fragment_shader.decompiled->constants)
-		{
-			const void *src = ucode + u32(constant.id / sizeof(*ucode));
+			auto ucode = (const rsx::fragment_program::ucode_instr *)info.fragment_shader.decompiled->raw->ucode_ptr;
 
-			const __m128i &vector = _mm_loadu_si128((const __m128i*)src);
-			const __m128i &shuffled_vector = _mm_shuffle_epi8(vector, mask);
-			_mm_stream_si128((__m128i*)dst, shuffled_vector);
+			auto dst = (const rsx::fragment_program::ucode_instr *)info.fragment_shader.const_buf.data();
 
-			if (0)
+			for (const auto& constant : info.fragment_shader.decompiled->constants)
 			{
-				float x = ((float*)dst)[0];
-				float y = ((float*)dst)[1];
-				float z = ((float*)dst)[2];
-				float w = ((float*)dst)[3];
+				const void *src = ucode + u32(constant.id / sizeof(*ucode));
 
-				LOG_WARNING(RSX, "fc%u = {%g, %g, %g, %g}", constant.id, x, y, z, w);
-			}
+				const __m128i &vector = _mm_loadu_si128((const __m128i*)src);
+				const __m128i &shuffled_vector = _mm_shuffle_epi8(vector, mask);
+				_mm_stream_si128((__m128i*)dst, shuffled_vector);
 
-			++dst;
+				if (1)
```

Comment:
This is just to test behaviour. Also the deletion above it is an error. This PR is not to be merged as is and is merely here to help with investigating the issue.